chore(blog): add release notes for 2.3 release

dalelane opened the pull request:
This pull request is a placeholder draft of the release notes for the upcoming 2.3 release.

Contributes to: asyncapi/spec#675

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
chore(blog): add release notes for 2.3 release

Contributes to: asyncapi/spec#675

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: placeholder sections for 2.3.0 release notes

Contributes to: asyncapi/spec#675

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: Add release notes for Solace bindings

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: add release notes for regex unicode compliance

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: release notes for channels now being optional

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: add release notes for allowing refs for servers and channels

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: release notes improvements

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

smoya commented on pages/blog/release-notes-2.3.0.md:
````diff
@@ -0,0 +1,84 @@
+For example, you could now have:
+
+```yaml
+PLACEHOLDER - asyncapi doc A without channels
````
We could use something like:

```yaml
asyncapi: 2.3.0
info:  
  description: Dictionary for all definitions of mycompany.com
components:  
  servers: 
    myserver:
      url: "http://localhost:5000/ws"
      protocol: was
  channels:
    myChannel1:
      description: "mychannel1"
    myChannel2:
      description: "mychannel2"
```

And then

```yaml
asyncapi: 2.3.0
info:  
  description: My awesome service API
servers: 
  myserver:
    $ref: 'dictionary.yaml#/components/servers/myserver'
channels:
  mychannel1:
    $ref: 'dictionary.yaml#/components/channels/mychannel1'
```

Commit:
Merge branch 'master' into 2022-01-release

dalelane committed:
docs: tidying up release notes

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
Merge branch '2022-01-release' of github.com:dalelane/website into 2022-01-release

boyney123 commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,116 @@
+channels:
+  mychannel1:
+    $ref: 'dictionary.yaml#/components/channels/myChannel'
+  mychannel1:
```
Hey @dalelane ,

Sorry maybe it's my misunderstanding here, but is `mychannel1` meant to be twice here for your channels example?

dalelane replied:
nope - nice catch, thanks :)

dalelane replied:
```suggestion
  mychannel2:
```

dalelane committed:
update pages/blog/release-notes-2.3.0.md

Co-authored-by: David Boyne <boyneyy123@gmail.com>

char0n commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,116 @@
+
+This improves compatibility with some JSON Schema parsers. For more details, check out the [pull request](https://github.com/asyncapi/spec-json-schemas/pull/145).
+
+
```
In light of https://github.com/asyncapi/spec/pull/696 I'm suggesting adding a following deprecation section:

```
## Deprecations

We're deprecating `Channel Item Object` `$ref` field starting from 2.3 version. The plan is to remove the `$ref` field 
from `Channel Item Object` in 3.x major version and allow `Reference Object` to be used wherever the `Channel Item Object` is defined.
```

dalelane committed:
chore: remove section on channels being optional

This change (from asyncapi/spec#661) is now being held back to
version 3.0 so I've removed the reference to it from the
release notes for 2.3

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: add release notes for channel item $ref deprecation

Contributes to: asyncapi/spec#607

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
chore: typo fix

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

char0n commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,92 @@
+
+## Deprecations
+
+The `$ref` field in [Channel Item Object](#https://www.asyncapi.com/docs/specifications/v2.3.0#channelItemObject) is deprecated from AsyncAPI 2.3.0, in favour of using [Reference Object](https://www.asyncapi.com/docs/specifications/v2.3.0#referenceObject) instead.
```
Hi @dalelane,

Would it be possible to rephrase this section? I've read it couple of times and the take away (in my understanding) is that the `Channel Item`.`$ref` field is deprecated and `Reference Object` should be used instead.

That's actually not the case. IMHO we should emphasize that though the `Channel Item`.`$ref` is now deprecated, it's still the only mechanism for referencing Channel Item Objects. `Reference Object` cannot be used in 2.3 version of spec to reference `Channel Item Object` as it would introduce ambiguity described in https://github.com/asyncapi/spec/issues/607#issuecomment-1018318910. Basically we're deprecating something that doesn't have other preferred way of substitution until 3.0 version.

In 3.0, the `Channel Object`.`$ref` will be dropped and the `Reference Object` will be allowed to be used wherever the `Channel Item Object` can. But this is not the case for version 2.3.

dalelane replied:
absolutely - thanks so much for checking it over and clarifying 👍

dalelane committed:
docs: address review comment

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

Commit:
Merge branch 'master' into 2022-01-release

derberg commented:
@alequetzalli anything from your side on this one?

derberg committed:
Merge branch 'master' into 2022-01-release

derberg commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,80 @@
+cover: /img/posts/release-notes-2.2.0/cover.webp
+authors:
+  - name: Dale Lane
+    photo: /img/avatars/questionmark.webp
```
it is intentional, right?

derberg reviewed (commented):
just pick different cover photo from https://unsplash.com/ and don't forget to pay credit to it at the bottom of the article like we do in other posts

dalelane committed:
docs: update cover image for release notes blog post

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: add avatar

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
chore: update timestamp for release notes blog post

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

dalelane committed:
docs: placeholder for tooling support updates

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

jonaslagoni commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,96 @@
+- [JavaScript Parser](https://github.com/asyncapi/parser-js/) uses latest **@asyncapi/specs** package and can be used to parse and validate 2.3.0 documents. Upgrade to 1.14.0 version.
+- [HTML template](https://github.com/asyncapi/html-template) uses the latest **@asyncapi/react-component** package. Upgrade to 0.23.0 version.
+- [JavaScript Converter](https://github.com/asyncapi/converter-js/) enables conversion from any AsyncAPI version into the 2.3.0 version of the spec. Upgrade to 0.7.0 version.
+
```
```suggestion
- [Modelina](https://github.com/asyncapi/modelina/) now also accepts AsyncAPI documents valid against the 2.3.0 version of the spec. Upgrade to 1.46.0 version.
```

dalelane committed:
docs: add Modelina to tooling support

Co-authored-by: Jonas Lagoni <jonas-lt@live.dk>

magicmatatjahu commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,96 @@
+- [JavaScript Converter](https://github.com/asyncapi/converter-js/) enables conversion from any AsyncAPI version into the 2.3.0 version of the spec. Upgrade to 0.7.0 version.
+- [Modelina](https://github.com/asyncapi/modelina/) now also accepts AsyncAPI documents valid against the 2.3.0 version of the spec. Upgrade to 1.46.0 version.
+
+Last but not least is the AsyncAPI Playground. Check new playground that uses latest HTML template and Markdown template with [this example](https://playground.asyncapi.io/?url=https://raw.githubusercontent.com/asyncapi/spec/v2.3.0/examples/websocket-gemini.yml).
```
```suggestion
Last but not least is the AsyncAPI Studio. Check new studio with [this example](https://studio.asyncapi.com/?url=https://raw.githubusercontent.com/asyncapi/spec/v2.3.0/examples/websocket-gemini.yml).
```

We are "slowly" moving from the playground to the studio :)

magicmatatjahu commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,96 @@
+The following official AsyncAPI tools are already updated to support 2.3.0 version of the specification:
+- JSON Schema that supports validation of AsyncAPI documents is updated in [this](https://github.com/asyncapi/asyncapi-node) repository. Also **@asyncapi/specs** package has been updated on NPM to version 2.13.0, and it contains the 2.3.0 JSON Schema.
+- [JavaScript Parser](https://github.com/asyncapi/parser-js/) uses latest **@asyncapi/specs** package and can be used to parse and validate 2.3.0 documents. Upgrade to 1.14.0 version.
+- [HTML template](https://github.com/asyncapi/html-template) uses the latest **@asyncapi/react-component** package. Upgrade to 0.23.0 version.
```
```suggestion
- [HTML template](https://github.com/asyncapi/html-template) uses the latest **@asyncapi/react-component** package. Upgrade to 0.24.7 version.
```

magicmatatjahu reviewed (commented):
Two improvements. I only just now noticed.

dalelane committed:
Update pages/blog/release-notes-2.3.0.md

Co-authored-by: Maciej Urbańczyk <urbanczyk.maciej.95@gmail.com>

dalelane committed:
Update pages/blog/release-notes-2.3.0.md

Co-authored-by: Maciej Urbańczyk <urbanczyk.maciej.95@gmail.com>

jonaslagoni commented on pages/blog/release-notes-2.3.0.md:
```diff
@@ -0,0 +1,96 @@
+- [HTML template](https://github.com/asyncapi/html-template) uses the latest **@asyncapi/react-component** package. Upgrade to 0.24.7 version.
+- [JavaScript Converter](https://github.com/asyncapi/converter-js/) enables conversion from any AsyncAPI version into the 2.3.0 version of the spec. Upgrade to 0.7.0 version.
+- [Modelina](https://github.com/asyncapi/modelina/) now also accepts AsyncAPI documents valid against the 2.3.0 version of the spec. Upgrade to 1.46.0 version.
+
```
```suggestion 
 - [Generator](https://github.com/asyncapi/generator/) uses the latest @asyncapi/parser package, so while generating output, it can validate 2.1.0 documents. Upgrade to 1.9.0 version.

```

quetzalliwrites reviewed (dismissed):
🛳

dalelane committed:
Update pages/blog/release-notes-2.3.0.md

Co-authored-by: Jonas Lagoni <jonas-lt@live.dk>

derberg committed:
Update release-notes-2.3.0.md

derberg commented:
/rtm
